sklearn/compose/_column_transformer.py

```diff
@@ -597,6 +608,50 @@ def _get_transformer_list(estimators):
     return transformer_list
 
 
+def select_types(dtypes):
+    """Generate a column selector callable (type-based) to be passed to
+    `make_column_transformer` or `ColumnTransformer` in place of the column
+    selections. The returned callable will return a boolean mask for column
+    selection when applied to a dataframe.
+
+    Parameters
+    ----------
+    dtypes : list of column dtypes to be selected from the dataset.
```

Comment:
Numpydoc: type on this line, semantics on the next

Reply:
I think we should make the same factory also able to select columns by nane. Please add a param to do so and rename the function